packages/core/src/index.ts

```diff
@@ -140,8 +143,16 @@ class Hopr<Chain extends HoprCoreConnector> extends EventEmitter {
     }
     this.bootstrapServers = options.bootstrapServers || []
     this.isBootstrapNode = options.bootstrapNode || false
-    this._interactions = new Interactions(this, this.mixer, (peer: PeerId) => this.network.networkPeers.register(peer))
-    this.network = new Network(this._libp2p, this._interactions)
```

Comment:
Why have you put this one level higher?

Reply:
part of my long-term incremental refactor to flatten out some of the code here - there is no need for a network `instance` - this simplifies the code.